submissions/TaniaBondarenko/frogger_game/app.js

```diff
@@ -0,0 +1,121 @@
+const borders = {
+    rightBorder: 400,
+    leftBorder: 5,
+    upBorder: 410,
+    downBorder: 70
+};
+
+const someGap = 10;
+
+class GameSprite {
+    constructor (x, y,sprite)  {
+        this.sprite = sprite;
+        this.x = x;
+        this.y = y;
+    }
+
+    render () {
+        ctx.drawImage(Resources.get(this.sprite), this.x, this.y);
+    };
+}
+
+class Enemy extends GameSprite {
+    constructor(x, y, sprite = "images/enemy-bug.png", speed = "75") {
+        super(x, y, sprite);
+        this.speed = speed;
+        this.player = player;
```

Comment:
`player` is a global variable.
Can you manage `this.player` the same way as `speed`, `x`, `y` are handled?

Reply:
If I put `player` in constructor of `Enemy` like this
`constructor(x,...,player) {
        ...
        this.player = player`,
and add player to new object Enemy (`new Enemy(5,60,player)`) - 
 then `this player` becomes some player, which is not defined in latest code, f.e. `this.player.x`.  (Cannot read property 'x' of undefine).
I dont understand how to  handle with this.

Reply:
@TaniaBondarenko 
That's because if Enemy constructor signature is `constructor(x, y, sprite = "images/enemy-bug.png", speed = "75", player)` and it expects x position as the first parameter, y position as the second, sprite string as the third, speed as the 4th and player as the 5th.
So when doing `new Enemy(5,60,player)` player is associated with `sprite` argument.

Reply:
Of course, thank you! Will fix it.

Reply:
Please check.